server/public/model/oauth_outgoing_connection_test.go

```diff
@@ -0,0 +1,313 @@
+package model
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	emptyString = ""
+	someString  = "userorpass"
+)
+
+func newValidOAuthOutgoingConnection() *OAuthOutgoingConnection {
+	return &OAuthOutgoingConnection{
+		Id:            NewId(),
+		CreatorId:     NewId(),
+		Name:          "Test Connection",
+		ClientId:      NewId(),
+		ClientSecret:  NewId(),
+		OAuthTokenURL: "https://nowhere.com/oauth/token",
+		GrantType:     "client_credentials",
+		CreateAt:      GetMillis(),
+		UpdateAt:      GetMillis(),
+		Audiences:     []string{"https://nowhere.com"},
+	}
+}
+
+func TestOAuthOutgoingConnectionIsValid(t *testing.T) {
+	var cases = []struct {
+		name   string
+		item   func() *OAuthOutgoingConnection
+		assert func(t *testing.T, oa *OAuthOutgoingConnection)
+	}{
+		{
+			name: "valid",
+			item: func() *OAuthOutgoingConnection {
+				return newValidOAuthOutgoingConnection()
+			},
+			assert: func(t *testing.T, oa *OAuthOutgoingConnection) {
+				require.Nil(t, oa.IsValid())
+			},
```

Comment:
Alternatively if you want to de-duplicate the `requireError` as before, you could define `requireError` and `requireNoError` as a variable at the top of the `TestOAuthOutgoingConnectionIsValid` function

```go
func TestOAuthOutgoingConnectionIsValid(t *testing.T) {
    requireError := func(t *testing.T, oa *OAuthOutgoingConnection) {
        // ...
    }

    var cases = []struct {
		name   string
		item   func() *OAuthOutgoingConnection
		assert func(t *testing.T, oa *OAuthOutgoingConnection)
	}{
```

Reply:
Yeah I know, but in the end there are just simple functions. I we do it we could probably make them general for all tests and replace then where appropriate.